core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java

```diff
@@ -918,21 +918,26 @@ public AuthenticationToken token() {
   @Override
   public synchronized void close() {
     if (closed.compareAndSet(false, true)) {
-      if (zooCacheCreated.get()) {
-        zooCache.get().close();
-      }
-      if (zooKeeperOpened.get()) {
-        zooSession.get().close();
```

Comment:
This needed to be moved to the end because it closes ZooKeeper, which deletes all ephemeral nodes and fires any watchers.